li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/metadataservice/MetadataServiceClient.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.metadataservice;
+
+import com.linkedin.kafka.clients.common.ClusterGroupDescriptor;
+
+import java.io.Closeable;
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.kafka.common.Configurable;
+
+
+// An interface to a generic metadata service which serves cluster and topic metadata for federated Kafka clusters.
+public interface MetadataServiceClient extends Configurable, Closeable {
```

Comment:
Autocloseable is better

Reply:
its generally better. since this is an Interface you should allow any error to be thrown not just IOException.

Reply:
Changed to AutoCloseable.